drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -406,55 +430,55 @@ static int rtc_stm32_init(const struct device *dev)
 	LL_RCC_ReleaseBackupDomainReset();
 #endif
 
-#if defined(CONFIG_COUNTER_RTC_STM32_CLOCK_LSI)
-
+	if (rtc_clock_freq == STM32_LSI_FREQ) {
 #if defined(CONFIG_SOC_SERIES_STM32WBX)
-	LL_RCC_LSI1_Enable();
-	while (LL_RCC_LSI1_IsReady() != 1) {
-	}
+		LL_RCC_LSI1_Enable();
+		while (LL_RCC_LSI1_IsReady() != 1) {
+		}
 #else
-	LL_RCC_LSI_Enable();
-	while (LL_RCC_LSI_IsReady() != 1) {
-	}
+		LL_RCC_LSI_Enable();
+		while (LL_RCC_LSI_IsReady() != 1) {
+		}
```

Comment:
Clocks activation should now be performed y clock_control driver and hence removed from here.